### PR TITLE
In-App Marketplace: Improve Product Card Clickability

### PR DESCRIPTION
**Pull request by Dan-Q**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Fixes [WCCOM#17831 "Ensure Product Card product titles are clickable"](https://github.com/Automattic/woocommerce.com/issues/17831) and [WCCOM#17792 "Make whole card clickable"](https://github.com/Automattic/woocommerce.com/issues/17792).

Design refinements to the In-App Marketplace indicate that:
- The entire product card, including but not limited to its title, should be clickable
  - However, clicking the vendor link on the product card should continue to point to the vendor's profile page
- Hovering over the product card should give appropriate feedback

This PR aims to achieve this.

### How to test the changes in this Pull Request:

1. Build WooCommerce Admin on your local environment e.g.: cd plugins/woocommerce-admin then nvm use && pnpm start.
2. Visit [WooCommerce > Marketplace > Extensions](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions&tab=extensions)
3. Test links to products and vendors, including their hover effect
4. See no `"can't embed <a> in <a>"` errors in your debug console

In the video below, watch the status bar in the bottom right to see that the link destination differs between the card and vendor link. Note that the double-encoded price issue is believed to be separate (something to do with [this WCCOM PR](https://github.com/Automattic/woocommerce.com/pull/17850), I presume). @poligilad-auto does this meet your expectations?

https://github.com/woocommerce/woocommerce/assets/53293/c58fc24d-a68e-48a5-93c0-fe7afaefebfe

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss [hunk 1]
```diff
@@ -6,6 +6,10 @@
 		border-radius: $grid-unit-05 !important;
 		margin-top: $large-gap;
 
+		&:hover {
+			outline: 1.5px solid var(--wc-blue);
```
Comment (andfinally) [inline plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss]: I don't know if we're using the gutenberg vars, but the [Figma](https://www.figma.com/file/2sXEw33oLyTZ7Mu3808sed/In-App-Marketplace?type=design&node-id=1960-21270&mode=dev) seems to have 

```scss
border: 1.5px solid var(--gutenberg-blue, #007CBA); 
```

which is a bit darker.

**Review by andfinally — APPROVED** [review]
Looks great, thanks! Just one slight question about the outline colour.

File: plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss [hunk 2]
```diff
@@ -6,6 +6,10 @@
 		border-radius: $grid-unit-05 !important;
 		margin-top: $large-gap;
 
+		&:hover {
+			outline: 1.5px solid var(--wp-admin-theme-color);
```
Comment (kdevnel) [inline plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss:10]: Thanks for this!

**Review by kdevnel — APPROVED** [review]
LGTM!

File: plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss [hunk 3]
```diff
@@ -56,11 +60,32 @@
 			text-overflow: ellipsis;
 		}
 
+		&__link {
+			&,
```
Comment (raicem) [inline plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss:64]: :exploding_head: Never occurred to me I can use the `&` by itself.

**Review by raicem — APPROVED** [review]

**Pull request merged**